transport-classes-io_uring/src/main/java/io/netty/channel/uring/Native.java

```diff
@@ -379,46 +384,50 @@ static RingBuffer createRingBuffer(int ringSize, int cqeSize, int setupFlags) {
         ObjectUtil.checkPositive(ringSize, "ringSize");
         ObjectUtil.checkPositive(cqeSize, "cqeSize");
         long[] values = ioUringSetup(ringSize, cqeSize, setupFlags);
-        assert values.length == 18;
+        assert values.length == 20;
         long cqkhead = values[0];
         long cqktail = values[1];
         int cqringMask = (int) values[2];
         int cqringEntries = (int) values[3];
-        long cqArrayAddress = values[4];
-        int cqringSize = (int) values[5];
-        long cqringAddress = values[6];
-        int cqringFd = (int) values[7];
-        int cqringCapacity = (int) values[8];
+        long cqkflags = values[4];
+        long cqArrayAddress = values[5];
+        int cqringSize = (int) values[6];
+        long cqringAddress = values[7];
+        int cqringFd = (int) values[8];
+        int cqringCapacity = (int) values[9];
         int cqeLength = (setupFlags & IORING_SETUP_CQE32) == 0 ? CQE_SIZE : CQE32_SIZE;
         CompletionQueue completionQueue = new CompletionQueue(
                 Buffer.wrapMemoryAddressWithNativeOrder(cqkhead, Integer.BYTES),
                 Buffer.wrapMemoryAddressWithNativeOrder(cqktail, Integer.BYTES),
                 cqringMask,
                 cqringEntries,
+                Buffer.wrapMemoryAddressWithNativeOrder(cqkflags, Integer.BYTES),
```

Comment:
The field is called `overflow` for the completion queue: https://www.mankier.com/2/io_uring_setup#:~:text=struct%20io_cqring_offsets,-{%20%20%20%20__u32

Having it be called "kflags" was a bit confusing when reading the PR.

Reply:
That's something different ;) overflow will be incremented in some cases... kflags is the one that has the flag. See also https://github.com/axboe/liburing/blob/bc8776af071656b47114d777b56f0e598431cf5d/src/queue.c#L44